### PR TITLE
FOGL-1459 - review plugin config reference in microservice.py

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Test result:
`============================================== 1163 passed, 31 skipped in 95.10 seconds ===============================================
`

### CODE REVIEW

File: python/foglamp/services/common/microservice.py [hunk 1]
```diff
@@ -50,15 +50,25 @@ class FoglampMicroservice(FoglampProcess):
...
 
-    def __init__(self, default_config):
+    def __init__(self):
         super().__init__()
         try:
+            # Configuration handled through the Configuration Manager
+            default_config = {
+                'local_services': {
+                    'description': 'Restrict microservice to localhost',
+                    'type': 'string',
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: boolean

default value False

so default will bind to 0.0.0.0

File: python/foglamp/services/common/microservice.py [hunk 2]
```diff
@@ -50,15 +50,25 @@ class FoglampMicroservice(FoglampProcess):
...
+                    'default': '127.0.0.1',
+                }
+            }
+
             loop = asyncio.get_event_loop()
 
-            # ----- Ref: FOGL-1155. We need to fetch host from configuration of plugin
-            category = self._name
+            # ----- Ref: FOGL-1155/1439.
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: we can change the base to 1432 or keep comment  of 1432

File: python/foglamp/services/south/server.py [hunk 3]
```diff
@@ -29,13 +29,7 @@ class Server(FoglampMicroservice):
...
 
     # Configuration handled through the Configuration Manager
-    _DEFAULT_CONFIG = {
-        'management_host': {
-            'description': 'Management host',
-            'type': 'string',
-            'default': '127.0.0.1',
-        }
-    }
+    _DEFAULT_CONFIG = {}
```
Comment (praveen-garg) [inline python/foglamp/services/south/server.py]: Please add a docstring, how  microservice .py + plugin config will be merged to it

File: python/foglamp/services/common/microservice.py (hunk 1 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/common/microservice.py]: Missed that before. Done.

File: python/foglamp/services/common/microservice.py (hunk 2 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/common/microservice.py]: Let us keep it independent of 1432.

File: python/foglamp/services/south/server.py (hunk 3 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/south/server.py]: It is known that keep_original_items = True merges the configuration.

**Comment by amarendra-dianomic** [comment]
@MarkRiddoch @praveen-garg feedback changes done. Please check.

File: python/foglamp/services/common/microservice.py [hunk 4]
```diff
@@ -67,7 +77,8 @@ def __init__(self, default_config):
             })
             self._core_microservice_management_client.create_configuration_category(config_payload)
             config = self._core_microservice_management_client.get_configuration_category(category_name=category)
-            host = config['management_host']['value']
+            is_local_services = True if config['local_services']['value'].lower() in ['true', 't', '1', 1] else False
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: ` if config['local_services']['value'].lower() == 'true'`, remove all other, it only creates confusion and we can add redundancy everywhere !

File: python/foglamp/services/common/microservice.py [hunk 5]
```diff
@@ -67,7 +77,8 @@ def __init__(self, default_config):
...
+            host = '127.0.0.1' if is_local_services is True else '0.0.0.0'
             # -----
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: remove this `# ---` block

File: python/foglamp/services/common/microservice.py (hunk 4 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/common/microservice.py]: All other values are perfectly valid for type "boolean", hence required.

File: python/foglamp/services/common/microservice.py [hunk 6]
```diff
@@ -67,7 +77,8 @@ def __init__(self, default_config):
             })
             self._core_microservice_management_client.create_configuration_category(config_payload)
             config = self._core_microservice_management_client.get_configuration_category(category_name=category)
-            host = config['management_host']['value']
+            is_local_services = True if config['local_services']['value'].lower() in ['true', 't', '1', 1] else False
+            host = '127.0.0.1' if is_local_services is True else '0.0.0.0'
```
[the remark this answers is not in the document]
Comment (amarendra-dianomic) [reply python/foglamp/services/common/microservice.py]: Let it be as it is as it adds more clarity.

File: python/foglamp/services/common/microservice.py (hunk 5 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/common/microservice.py]: done

File: python/foglamp/services/south/server.py (hunk 3 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/south/server.py]: done

**Comment by amarendra-dianomic** [comment]
@praveen-garg feedback changes done. Pl check.

File: python/foglamp/services/common/microservice.py (hunk 2 above)
Comment (praveen-garg) [reply python/foglamp/services/common/microservice.py]: please remove this and change the base to 1432.

File: python/foglamp/services/common/microservice.py [hunk 7]
```diff
@@ -67,8 +77,8 @@ def __init__(self, default_config):
             })
             self._core_microservice_management_client.create_configuration_category(config_payload)
             config = self._core_microservice_management_client.get_configuration_category(category_name=category)
-            host = config['management_host']['value']
-            # -----
+            is_local_services = True if config['local_services']['value'].lower() in ['true', 't', '1', 1] else False
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: `config['local_services']['value'].lower() in ['true']`

File: python/foglamp/services/common/microservice.py [hunk 8]
```diff
@@ -67,8 +77,8 @@ def __init__(self, default_config):
...
+            host = '127.0.0.1' if is_local_services is True else '0.0.0.0'
```
Comment (praveen-garg) [inline python/foglamp/services/common/microservice.py]: if is_local_services is True IS more concise and equivalent to `if is_local_services`

**Comment by amarendra-dianomic** [comment]
@praveen-garg feedback changes done. Pl check.

File: python/foglamp/services/common/microservice.py [hunk 9]
```diff
@@ -67,8 +77,8 @@ def __init__(self, default_config):
             })
             self._core_microservice_management_client.create_configuration_category(config_payload)
```
Comment (ashish-jabble) [inline python/foglamp/services/common/microservice.py]: L#68 comment move here and Add below code line for parent/child relationship and comment it, so post merge of 1432 below code works
> self._core_microservice_management_client.create_child_category("General", ["Security"])

**Pull request merged**